app/scripts/metamask-controller.js

```diff
@@ -493,11 +494,34 @@ export default class MetamaskController extends EventEmitter {
     this.selectedNetworkController = new SelectedNetworkController({
       messenger: this.controllerMessenger.getRestricted({
         name: 'SelectedNetworkController',
-        allowedActions: ['NetworkController:getNetworkClientById'],
+        allowedActions: [
+          'NetworkController:getNetworkClientById',
+          'NetworkController:getState',
+          `PermissionController:hasPermissions`,
+        ],
         allowedEvents: ['NetworkController:stateChange'],
       }),
+      state: {
+        ...initState.SelectedNetworkController,
+        perDomainNetwork:
+          this.preferencesController.store.getState().useRequestQueue,
+      },
+    });
+
+    // couples the useRequestQueue featureflag with the perDomainNetwork feature flag
+    this.preferencesController.store.subscribe(({ useRequestQueue }) => {
+      if (
+        useRequestQueue !==
+        this.selectedNetworkController.state.perDomainNetwork
+      ) {
+        this.selectedNetworkController.setPerDomainNetwork(useRequestQueue);
+      }
```

Comment:
we should probably just remove this redundant `perDomainNetwork` state from the `SelectedNetworkController` and just query for the `useRequestQueue` state on the `PreferencesController` wherever we need it. But this will require another update to the `SelectedNetworkController` and should be done in a separate PR.

Reply:
This dedupe is happening here btw: https://github.com/MetaMask/core/pull/3989